modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -261,7 +271,18 @@ public static void removeAcceleratorsFromScene(List<? extends MenuItem> items, N
         removeAcceleratorsFromScene(items, scene);
     }
 
-    public static void removeAcceleratorsFromScene(List<? extends MenuItem> items, Scene scene) {
+    public static void removeAcceleratorsFromScene(ObservableList<? extends MenuItem> items, Scene scene) {
+        WeakReference<ListChangeListener<MenuItem>> listenerW = menuListChangeListenerMap.get(new IdentityWrapperListChangeListener(items));
+        if (listenerW != null) {
+            ListChangeListener<MenuItem> listChangeListener = listenerW.get();
+            if (listChangeListener != null) {
+                items.removeListener(listChangeListener);
+            }
+        }
+        removeAcceleratorsFromScene((List<MenuItem>)items, scene);
+    }
+
+    private static void removeAcceleratorsFromScene(List<? extends MenuItem> items, Scene scene) {
```

Comment:
May I suggest a change name: removeAcceleratorsFromScenePrivate() to avoid confusion?

Reply:
I'm not in favor of using `Private` in a method name. That is clear from the method signature and overloading methods is a valid choice  In my opinion, this is fine as is. 
But we could also think about naming it: `removeAcceleratorsFromSceneImpl()`

Reply:
I've been working on changes for a possible follow-up PR which address more bugs. For adding accelerators, there are 3 public methods, for removing there are currently 4.

I've looked at it, and it can/should be made to have 3 public remove methods that exactly mirror the public add methods (if you use a specific one to add then use the corresponding one to remove).

Further, for every private addAcceleratorsFromScene method and doAcceleratorInstall method I believe there should be a private corresponding method which reverses it. I had to rename a couple private removeAcceleratorsFromScene to doAcceleratorUninstall.

So yes, this is a very confusing class. Pairing up the add/remove methods make sense to me. Once that's done, we might want to rename some of the private ones just so it's easier to understand what each one is doing, but the public ones are fine I think.

Reply:
I just want to avoid confusion when we have public and non-public methods with the same name.  but it's a minor comment, especially if there will be subsequent rework later.

Reply:
We use this pattern of having methods with the same name and overloaded args, some of which are not public, in many places. In general this seems fine. In specific cases where it might help alleviate confusion, using "Impl" as a suffix is OK (and better than "Private" since we might make some impl methods package scope).